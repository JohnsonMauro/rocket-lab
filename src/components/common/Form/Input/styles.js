import styled from '@emotion/styled';
import { css } from '@emotion/core';

const Error = css`
  background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' width='12' height='12' fill='none' stroke='%23dc3545' viewBox='0 0 12 12'%3e%3ccircle cx='6' cy='6' r='4.5'/%3e%3cpath stroke-linejoin='round' d='M5.8 3.6h.4L6 6.5z'/%3e%3ccircle cx='6' cy='8.2' r='.6' fill='%23dc3545' stroke='none'/%3e%3c/svg%3e");
  background-repeat: no-repeat;
  background-position: 98% 50%;
  background-size: 16px;
`;

const InputStyles = styled.input`
  ${({ error, touched }) => error && touched && Error};
  border: 0.2px solid
    ${({ error, theme, touched }) =>
      error && touched ? `${theme.colors.error}` : `${theme.colors.gainsboro}`};
  border-radius: 1.8px;
  display: block;
  height: 34px;
  max-width: 332px;
  outline: none;
  padding: 6px 5px;
  width: 100%;

  &::placeholder {
    color: ${({ theme }) => theme.colors.gainsboro};
  }
`;

export { InputStyles };
