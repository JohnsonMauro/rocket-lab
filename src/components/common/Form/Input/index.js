import ErrorMessage from '../Message/ErrorMessage';
import { InputStyles } from './styles';

const Input = ({
  className,
  error,
  id,
  maxLength,
  minLength,
  name,
  onChange,
  placeholder,
  required,
  type,
  value,
}) => (
  <>
    <InputStyles
      className={className}
      error={error}
      id={id}
      maxLength={maxLength}
      minLength={minLength}
      name={name}
      placeholder={placeholder}
      onChange={onChange}
      required={required}
      type={type}
      value={value}
    ></InputStyles>
    {!!error && <ErrorMessage>{error}</ErrorMessage>}
  </>
);

export default Input;
