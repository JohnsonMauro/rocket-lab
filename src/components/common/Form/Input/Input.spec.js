import React from 'react';
import { render } from '@testing-library/react';

import GlobalsContainer from '@test/GlobalsContainer';
import Input from './index';

describe('<Input />', () => {
  it('Renders the Input component', () => {
    const { container } = render(<Input />, {
      wrapper: GlobalsContainer,
    });
    expect(container).toMatchSnapshot();
  });
});
