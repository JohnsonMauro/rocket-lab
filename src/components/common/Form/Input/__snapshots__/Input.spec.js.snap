// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Renders the Input component 1`] = `
<div>
  <input
    class="css-rgdr6w-InputStyles emzn0i0"
    value=""
  />
</div>
`;
